examples/gno.land/p/moul/errs/errs.gno

```diff
@@ -0,0 +1,139 @@
+// Package errs provides utilities for combining multiple errors.
+// This is a simplified version of https://github.com/uber-go/multierr (MIT License),
+// adapted for the Gno programming language with a focus on core functionality
+// and idiomatic usage patterns.
+//
+// Example usage:
+//
+//	err1 := doSomething()
+//	err2 := doSomethingElse()
+//	if err := errs.Combine(err1, err2); err != nil {
+//	    return err // Returns combined errors or single error
+//	}
+package errs
+
+import (
+	"strings"
+)
+
+// multiError represents multiple errors combined into one.
+type multiError struct {
+	errors []error
+}
+
+// Error implements the error interface by returning a single-line representation
+// of all contained errors, separated by semicolons.
+func (m *multiError) Error() string {
+	if m == nil || len(m.errors) == 0 {
+		return ""
+	}
+
+	var b strings.Builder
+	first := true
+	for _, err := range m.errors {
+		if first {
+			first = false
+		} else {
+			b.WriteString("; ")
+		}
+		b.WriteString(err.Error())
+	}
+	return b.String()
+}
+
+// String returns a multi-line representation of the error.
+func (m *multiError) String() string {
+	if m == nil || len(m.errors) == 0 {
+		return ""
+	}
+
+	var b strings.Builder
+	b.WriteString("the following errors occurred:")
+	for _, err := range m.errors {
+		b.WriteString("\n -  ")
+		b.WriteString(err.Error())
+	}
+	return b.String()
+}
+
+// Errors returns the slice of underlying errors contained in this multiError.
+// Returns nil if the receiver is nil.
+func (m *multiError) Errors() []error {
```

Comment:
`Unwrap() []error`?

We don't have the extra functions in `errors`, but I think we should in the future, post-reflection